tasks/kernel_matrix_testing/download.py

```diff
@@ -83,16 +92,20 @@ def download_rootfs(ctx, rootfs_dir, backup_dir, revert=False):
             debug(f"[debug] updating {f} from S3.")
             to_download.append(f)
 
+    if len(to_download) == 0:
+        warn("[-] No update required for rootfs images")
+        return
+
     # download files to be updates
     fd, path = tempfile.mkstemp()
     try:
         with os.fdopen(fd, 'w') as tmp:
```

Comment:
A suggestion for future improvement, you can use `with tempfile.TemporaryFile() as tmp:` rather than `tempfile.mkstemp+os.fdopen+os.remove`